src/Fixer/SpecifyArgSeparatorFixer.php

```diff
@@ -0,0 +1,149 @@
+            ],
+            'Function `http_build_query()` uses `arg_separator.output` ini directive as default argument separator, '
+            . 'however when its default value "&" is changed, query string assembled by the method will be '
+            . 'unexpectedly invalid. This Fixer forces you to not rely on ini settings and rather define '
+            . '`$arg_separator` in third parameter of the function.',
+            null,
+            null,
+            'Risky when other than default "&" argument separator should be used in query strings.'
+        );
+    }
+
+    public function isCandidate(Tokens $tokens): bool
+    {
+        return $tokens->isTokenKindFound(T_STRING);
+    }
+
+    public function isRisky(): bool
+    {
+        return true;
+    }
+
+    public function fix(\SplFileInfo $file, Tokens $tokens): void
+    {
+        foreach ($tokens as $index => $token) {
+            if ($token->isGivenKind(T_STRING) && $token->getContent() === 'http_build_query') {
+                $previousIndex = $tokens->getPrevMeaningfulToken($index);
+
+                // Function imports and function definitions should not be affected
+                if ($previousIndex !== null
+                    && ($tokens[$previousIndex]->isGivenKind(CT::T_FUNCTION_IMPORT)
+                        || $tokens[$previousIndex]->isGivenKind(T_FUNCTION))) {
+                    return;
+                }
+
+                $this->fixFunction($tokens, $index);
+            }
+        }
+    }
+
+    public function getName(): string
+    {
+        return mb_strtolower(self::class);
+    }
+
+    public function getPriority(): int
+    {
+        return 0;
+    }
+
+    public function supports(\SplFileInfo $file): bool
+    {
+        return true;
+    }
+
+    private function fixFunction(Tokens $tokens, int $functionIndex): void
+    {
+        $startBraceIndex = $tokens->getNextTokenOfKind($functionIndex, ['(']);
+        if ($startBraceIndex === null) {
+            return;
+        }
+
+        $endBraceIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $startBraceIndex);
+        $ampToken = new Token([T_STRING, "'&'"]);
+
+        $commaIndexes = [];
+
+        for ($index = $startBraceIndex + 1; $index < $endBraceIndex; ++$index) {
+            $token = $tokens[$index];
+
+            // Parameter surrounded by brace, skip to its end
+            if ($token->equals('(')) {
+                $index = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
+
+                continue;
+            }
+
+            // Array parameter, skip to its end
+            if ($token->isGivenKind(CT::T_ARRAY_SQUARE_BRACE_OPEN)) {
+                $index = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_ARRAY_SQUARE_BRACE, $index);
+
+                continue;
+            }
+
+            if ($token->equals(',')) {
+                $commaIndexes[] = $index;
+
+                continue;
+            }
+        }
```

Comment:
Might be helpful:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.11/src/Tokenizer/Analyzer/ArgumentsAnalyzer.php

Reply:
I'd be happy to check after that, there is lot of outsourcable code now